firefox-ios/Client/Frontend/Home/Homepage Rebuild/Redux/HomepageMiddleware.swift

```diff
@@ -48,4 +54,84 @@ final class HomepageMiddleware {
             break
         }
     }
+
+    // MARK: - Notifications
+    private func observeNotifications() {
+        let notifications: [Notification.Name] = [
+            UIApplication.didBecomeActiveNotification,
+            .FirefoxAccountChanged,
+            .PrivateDataClearedHistory,
+            .ProfileDidFinishSyncing,
+            .TopSitesUpdated,
+            .DefaultSearchEngineUpdated,
+            .BookmarksUpdated,
+            .RustPlacesOpened
+        ]
+
+        notifications.forEach {
+            notificationCenter.addObserver(
+                self,
+                selector: #selector(handleNotifications),
+                name: $0,
+                object: nil
+            )
+        }
+    }
+
+    @objc
+    private func handleNotifications(_ notification: Notification) {
+        // This update occurs independently of any specific window, so for now we send `.unavailable`
+        // as the window UUID. Reducers responding to these types of messages need to use care not to
+        // propagate that UUID in any subsequent actions or state changes.
+        switch notification.name {
+        case UIApplication.willEnterForegroundNotification:
+            let storiesAction = HomepageAction(
+                windowUUID: .unavailable,
+                actionType: HomepageMiddlewareActionType.enteredForeground
```

Comment:
iffy on the naming, since the action taken was `enteredForeground`, but for the others I used `xxUpdated` because there are various actions and wanted to simply the calls for now. Maybe can rename when needed?